File: 80.remove-duplicates-from-sorted-array-ii.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Solution80 {
    public int RemoveDuplicates(int[] nums) {
        if(nums == null || nums.Length == 0) return 0;
        int start = 0;
        int prev = nums[0];
        int cnt = 1;
        for(int i = 1; i < nums.Length; ++i)
        {
            if(nums[i] != prev)
            {
                start ++;
                nums[start] = nums[i];
                prev = nums[start];
                cnt = 1;
            }
            else if(cnt == 1)
            {
                start ++;
                nums[start] = nums[i];
                cnt ++;
            }
        }

        return start +1;
    }
}

File: 80_remove-duplicates-from-sorted-array-ii.py
from typing import List


class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        m = len(nums)
        if m <=2: return m

        start = 0
        prev = nums[0]
        cnt = 1
        for i in range(1, m):
            if nums[i] != prev:
                start +=1
                nums[start] = prev = nums[i]
                cnt = 1
            elif cnt == 1:
                start +=1
                nums[start] = nums[i]
                cnt += 1
        
        return start + 1
        